### PR TITLE
Readonly client on brand retail residential and friend

### PR DESCRIPTION
**Pull request by danigargar**
Modified back and front to make client readonly on Friends, RetailAccounts and ResidentialDevices edition

#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, portal/platform, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/Friend/Friend.php [hunk 1]
```diff
@@ -228,6 +229,21 @@ public function setRuriDomain(?string $ruriDomain = null): static
...
+     * {@inheritDoc}
+     */
+    public function setCompany(CompanyInterface $company): static
+    {
+        if (!$this->isNew()) {
+            Assertion::eq(
+                $this->getInitialValue('companyId'),
+                $company->getId()
+            );
+        }
```
Comment (Kaian) [inline library/Ivoz/Provider/Domain/Model/Friend/Friend.php:242]: This is ok but other entities have this implemented using a lifecyle service. See https://github.com/irontec/ivozprovider/blob/bleeding/library/Ivoz/Provider/Domain/Service/FixedCostsRelInvoice/AvoidUpdates.php

I don't know if there are other entites that implement this check in the setter. If these are the first ones, I would prefer to stick to the previous implemented mechanisms and create an AvoidCompanyUpdate.php service for these entities.

**Review by Kaian — CHANGES_REQUESTED** [review]

**Review by Kaian — CHANGES_REQUESTED** [review]
If it makes sense, we can add some testing for these new services like [this one](https://github.com/irontec/ivozprovider/blob/cc9672ca31848755de77efa868d8d9968cc93b04/library/spec/Ivoz/Provider/Domain/Service/FixedCostsRelInvoice/AvoidUpdatesSpec.php#L10)

File: library/Ivoz/Provider/Domain/Service/Friend/AvoidUpdateCompany.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace Ivoz\Provider\Domain\Service\Friend;
+
+use Ivoz\Core\Domain\Assert\Assertion;
+use Ivoz\Core\Domain\Service\AvoidEntityUpdatesAbstract;
+use Ivoz\Provider\Domain\Model\Friend\FriendDto;
+use Ivoz\Provider\Domain\Model\Friend\FriendInterface;
+use Ivoz\Provider\Domain\Service\Friend\FriendLifecycleEventHandlerInterface;
+
+class AvoidUpdateCompany implements FriendLifecycleEventHandlerInterface
```
[the remark this answers is not in the document]
Comment (Kaian) [reply library/Ivoz/Provider/Domain/Service/Friend/AvoidUpdateCompany.php:11]: Ok I see, assertUnchanged does not what I expected. This implementation is better.

[the remark this answers is not in the document]
Comment (danigargar) [reply library/Ivoz/Provider/Domain/Service/Friend/AvoidUpdateCompany.php:11]: assertUnchanged checks if  an Entity has been changed, but for a set fields (second param) which are ignored.

In this case I only need to check for company and it looked harder to maintain.

**Review by Kaian — APPROVED** [review]

**Pull request merged**